kolibri/core/assets/src/views/sync/SelectAddressModalGroup/useConnectionChecker.js

```diff
@@ -0,0 +1,96 @@
+import { ref, onBeforeUnmount } from 'kolibri.lib.vueCompositionApi';
+import { get, set, useMemoize, useTimeoutPoll } from '@vueuse/core';
+import { updateConnectionStatus } from './api';
+import { ConnectionStatus } from './constants';
+
+function useMemoizeWithExpiry(asyncFunction, options = {}) {
+  const getKey = args => (options.getKey ? options.getKey(...args) : args[0]);
+  const memoizedFunc = useMemoize(asyncFunction, options);
+
+  return async (...args) => {
+    let result = null;
+    try {
+      result = await asyncFunction(...args);
+      setTimeout(() => memoizedFunc.cache.delete(getKey(args)), options.expiry || 10000);
+    } catch (e) {
+      // clear immediately
+      memoizedFunc.cache.delete(getKey(args));
+    }
+    return result;
+  };
+}
+
+/**
+ *
+ * @param {Ref<NetworkLocation[]>} devices
+ * @param {Number} threshold - minimum `since_last_accessed` needed to perform check
+ * @param {Number} interval - time between running individual checks
+ * @param {Number} concurrency - the number of simultaneous connection checks
+ * @return {{
+ *  checkFailed: Ref<bool>, isChecking: Ref<bool>, hasChecked: Ref<bool>,
+ *  doCheck: (function(id: string): Promise<NetworkLocation>),
+ * }}
+ */
+export default function useConnectionChecker(
+  devices,
+  { threshold = 5, interval = 2, concurrency = 3 } = {}
```

Comment:
This updates the connection status when keeping it refreshed is important, like the user sitting with the network device modal open.

With these parameters, I tried to balance the situation where there might be many devices on the network. Open to tweaking these